amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/auth/KclStsAssumeRoleCredentialsProvider.java

```diff
@@ -0,0 +1,53 @@
+package software.amazon.kinesis.multilang.auth;
+
+import java.util.Arrays;
+
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
+import software.amazon.kinesis.multilang.NestedPropertyKey;
+import software.amazon.kinesis.multilang.NestedPropertyProcessor;
+
+public class KclStsAssumeRoleCredentialsProvider implements AwsCredentialsProvider, NestedPropertyProcessor {
```

Comment:
SDK v1 STSAssumeRoleSessionCredentialsProvider [worked very differently](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-client-credentials.html). Old STSAssumeRoleSessionCredentialsProvider builder could adjust properties and rebuild using the same builder. 

New StsAssumeRoleCredentialsProvider must be built using a STSClient and AssumeRoleRequest. If the properties of these change, StsAssumeRoleCredentialsProvider must be rebuilt.

Restructured this class with a config and factory to construct new StsAssumeRoleCredentialsProvider whenever resolveCredentials is called.

Reply:
Can't we just have builders for sts client and AssumeRoleRequest as class member and populate them on the fly?